BUILD

```diff
@@ -0,0 +1,17 @@
+package(default_visibility = ["//visibility:public"])
+load(":merge.bzl", "merge")
+
+merge(
+    name = "merge",
+    out = "asmcli",
+    chunks = glob(
+        ["asmcli/**/*.sh", "asmcli/asmcli"],
+        exclude = ["**/test*/**"],  # exclude any directory that starts with test
+    ),
+    merge_tool = "//:merge_script"
+)
```

Comment:
Is there a way to do this without needing a custom Bazel rule? It feels like we should be able to get away with just a `sh_binary` rule called `merge`.

Reply:
I looked around more and I don't think `sh_binary` would work since the native build targets are used for building or compiling, whereas what we want to do here is actually executing an executable (any executable not limited to bash). If we build a `sh_binary`, all it does is bring all dependencies to the `bazel-bin` directory. Shell does not need additionally building since we could just execute that. 

Similarly in other languages (`Java`, `Golang`, etc.), the process of building a target is generating an executable, but what we want to do here is to *execute* an executable. I followed this [example](https://github.com/bazelbuild/examples/tree/master/rules/actions_run), but I think in additions to `ctx.actions.run`, we could also use `ctx.actions.executable`. 

In summary, I think having a custom rule is necessary to fit in our goal, but if ever we found out that a single `sh_binary` would work, I will be responsible for switching to using that, which shouldn't be too hard. Does this sound reasonable?

Reply:
Updated to use `genrule`